api/app/routes/admin.py

```
@@ -277,12 +307,227 @@ async def get_metrics():
+
+
 @router.get("/faqs", response_model=FAQListResponse)
-async def get_all_faqs_for_admin_route():
-    """Get all FAQs for the admin interface."""
-    logger.info("Admin request to fetch all FAQs")
-    faqs = faq_service.get_all_faqs()
-    return FAQListResponse(faqs=faqs)
+async def get_all_faqs_for_admin_route(
+    page: int = 1,
+    page_size: int = 10,
+    search_text: str = None,
```

Ruff (0.13.1)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)